### DISCUSSION TITLE
[Share] add/remove one tag to/from current item by one action  只用一个动作 就能完成 删除或者添加 某个特定标签 的功能

### DISCUSSION THREAD

**Discussion by Fangqi980214991015** [windingwind/zotero-actions-tags#429] [Action Scripts]
### Description

using zotero-action-tags to add or remove tags, one has to add a tag and remove the same one separately.i create a simple script to add/remove one tag to/from current item by one action.  people accustomed to getting tags by zotero-tag can use this script to act as it.

### Event

None

### Operation

Script

### Data

```javascript
if (!item) return;

// 规定要移除或者添加的 目标标签 这里替换成任何你要快捷键添加或者删除的标签
// replace read with any tab you want 
const targettag = "read"
// 获取现在选中条目的 所有现有标签
const tags = item.getTags();
// 比对 现有标签 中是否含有 目标标签
const hasttag = tags.find(t => t.tag === targettag);

if (hasttag) {
    // 如果含有 目标标签，则移除目标标签
	item.removeTag(targettag);
    return `Remove tag `+ targettag + ` from item ${item.getField("title")}`;
} else {
    // 如果没有 目标标签，则添加目标标签
    item.addTag(targettag, 1);
    return `Add tag `+ targettag + ` to item ${item.getField("title")}`;
}
```

### Anything else

Just customize the shortcut keys and names according to your needs
快捷键和名称按照自己的需求自定义就行

**Comment by windingwind** [comment]
不用这么复杂，直接使用operation=toggle tags即可。
